app/src/main/java/org/p2p/wallet/sell/ui/payload/SellPayloadPresenter.kt

```diff
@@ -88,9 +136,61 @@ class SellPayloadPresenter(
         val moonpayUrl = moonpayWidgetUrlBuilder.buildSellWidgetUrl(
             tokenSymbol = Constants.SOL_SYMBOL,
             userAddress = userAddress,
-            fiatSymbol = "usd",
-            tokenAmountToSell = "0.005",
+            fiatSymbol = fiat?.symbol.orEmpty(),
+            tokenAmountToSell = userSelectedAmount.toString(),
         )
         view?.showMoonpayWidget(url = moonpayUrl)
     }
+
+    override fun onTokenAmountChanged(newValue: String) {
+        val newDoubleValue = newValue.toBigDecimalOrZero()
+        sellQuoteJob?.cancel()
+        val buttonState = when {
+            newDoubleValue.isLessThan(minSellAmount) -> {
+                CashOutButtonState(
+                    isEnabled = false,
+                    backgroundColor = R.color.bg_rain,
+                    textColor = R.color.text_mountain,
+                    text = resourceProvider.getString(R.string.sell_min_sol_amount, minSellAmount.toString())
+                )
+            }
+            newDoubleValue.isMoreThan(maxSellAmount.orZero()) -> {
+                CashOutButtonState(
+                    isEnabled = false,
+                    backgroundColor = R.color.bg_rain,
+                    textColor = R.color.text_mountain,
+                    text = resourceProvider.getString(R.string.sell_max_sol_amount, maxSellAmount.toString())
+                )
+            }
+            newDoubleValue.isMoreThan(userSolToken?.total.orZero()) -> {
+                CashOutButtonState(
+                    isEnabled = false,
+                    backgroundColor = R.color.bg_rain,
+                    textColor = R.color.text_mountain,
+                    text = resourceProvider.getString(R.string.sell_not_enough_sol)
+                )
+            }
+            else -> {
+                userSelectedAmount = newDoubleValue
+                startLoadSellQuoteJob()
+                CashOutButtonState(
+                    isEnabled = true,
+                    backgroundColor = R.color.bg_night,
+                    textColor = R.color.text_snow,
+                    text = resourceProvider.getString(R.string.common_cash_out)
+                )
+            }
+        }
+        view?.reset().takeIf { !buttonState.isEnabled }
```

Comment:
reset would be called anyways, take if doesn't work such way

Reply:
yeah fixed it